### PR TITLE
add min and max to jsonAll export #275

### PR DESCRIPTION
**Pull request by ebroecker**
fixes 
* min and max for json import/export
* add tests for this
* remove import * from canmatrix

### CODE REVIEW

File: src/canmatrix/cmjson.py [hunk 1]
```diff
@@ -211,13 +213,20 @@ def load(f, **options):
...
+                newsignal = canmatrix.Signal(signal["name"],
                                    startBit=signal["start_bit"],
                                    size=signal["bit_length"],
                                    is_little_endian=is_little_endian,
                                    is_signed=is_signed,
                                    factor=signal["factor"],
                                    offset=signal["offset"])
+
+                if "min" in signal and signal["min"]:
+                    newsignal.min = decimal.Decimal(signal["min"])
```
Comment (ebroecker) [inline src/canmatrix/cmjson.py]: @altenky 
I hast to add "decimal.Decimal"-casting here, 
not sure why ` min = attr.ib(` from signal-class did not work.
What do I wrong here?

Comment (altendky) [reply src/canmatrix/cmjson.py]:
```suggestion
                    newsignal.min = newsignal.float_factory(signal["min"])
```

The converter for an `attr.ib()` is only applied when set through the constructor.  We could make `@property`s which apply the converter on assignment as well but, I dunno...  I do have a thing which does that automatically.  But, if we want that it should certainly be it's own effort.

Do beware when writing json as to whether you write floats or strings.  The former loses the benefit of using `decimal.Decimal()`.

Sorry I missed the review request...

File: src/canmatrix/cmjson.py [hunk 2]
```diff
@@ -199,32 +200,42 @@ def load(f, **options):
                 newframe.extended = 0
 
             for signal in frame["signals"]:
-                if "is_big_endian" in signal and signal["is_big_endian"]:
+                if signal.get("is_big_endian", False):
```
Comment (altendky) [inline src/canmatrix/cmjson.py]:
```suggestion
                is_little_endian = signal.get("is_big_endian", False)
```

While we're refactoring...  :]  Do note that these are both a little different than before, though maybe more what we want, maybe.  In the left-side code the result was `True` if the value wasn't specified.  Not sure if you want to default to little or big endian.

File: src/canmatrix/cmjson.py [hunk 3]
```diff
@@ -199,32 +200,42 @@ def load(f, **options):
...
                                    startBit=signal["start_bit"],
                                    size=signal["bit_length"],
                                    is_little_endian=is_little_endian,
                                    is_signed=is_signed,
+                                   is_float = is_float,
                                    factor=signal["factor"],
                                    offset=signal["offset"])
-                if "unit" in signal and signal["unit"]:
+
+                if signal.get("min", False):
```
Comment (altendky) [inline src/canmatrix/cmjson.py]:
```suggestion
                if signal.get("min") is not None:
```

What about a `0` minimum?  It'll be falsey...

**Pull request merged**